code/mob/living/critter/ai/shared.dm

```diff
@@ -131,14 +85,18 @@
 	var/max_path_dist = 50 //keeping this low by default, but you can override it - see /datum/aiTask/sequence/goalbased/rally for details
 	var/list/found_path = null
 	var/atom/move_target = null
-	var/move_through_space = FALSE
 
 // use the target from our holder
 /datum/aiTask/succeedable/move/proc/get_path()
 	if(!move_target)
 		fails++
 		return
-	src.found_path = get_path_to(holder.owner, move_target, src.max_path_dist, 0, null, !move_through_space)
+	if(length(holder.target_path) && holder.target_path[length(holder.target_path)] == move_target)
+		src.found_path = holder.target_path
+	else
+		src.found_path = get_path_to(holder.owner, move_target, src.max_path_dist, 0, null, !move_through_space)
+		if(get_turf(holder.target) == move_target)
```

Comment:
I'm not sure what this check is for but I assume it's necessary

Reply:
in case someone wants to target a thing with an ai task but move somewhere else, ie shooting - target a dude, move a few tiles away so you can shoot 'em.